src/sidebars/search/search-controller.js

```diff
@@ -21,7 +21,12 @@
         }
 
         select(stock) {
-            this.selectionService.select(stock);
+            if (!this.store) {
+                this.store = window.storeService.open(window.name);
+            }
+            if (!this.store.isCompact() && stock.favourite) {
+                this.selectionService.select(stock);
+            }
```

Comment:
Direct selection of search results still seems weird to me. It also seems to reduce the usefulness of having favourites at all. So I can search for something, select it, then I go back to the regular view look at the chart. I then select one of my favourites to look at, but then want to go back to the other chart that I was just viewing, if I forget the name of the stock I had selected from search, then there's no way to go back to that chart because the stock is not favourited. Seems like potential for bad UX to me. I'm happy to change it though, if I'm in the minority in that opinion.